test/util.js

```diff
@@ -1,55 +1,101 @@
 'use strict';
 
-const fsMock = require('fs');
 const path = require('path');
+const del = require('del');
+const makeDir = require('make-dir');
+const parentModule = require('parent-module');
+const os = require('os');
+const fs = require.requireActual('fs');
 
-exports.absolutePath = str => path.join(__dirname, str);
+class TempDir {
+  dir: string;
 
-exports.mockStatIsDirectory = function mockStatIsDirectory(result) {
-  const stats = {
-    isDirectory: () => result,
-  };
+  constructor() {
+    /**
+     * Get the actual path for temp directories that are symlinks (MacOS).
+     * Without the actual path, tests that use process.chdir will unexpectedly
+     * return the real path instead of symlink path.
+     */
+    const tempDir = fs.realpathSync(os.tmpdir());
+    /**
+     * Get the pathname of the file that imported util.js.
+     * Used to create a unique directory name for each test suite.
+     */
+    const parent = parentModule();
+    const relativeParent = path.relative(process.cwd(), parent);
 
-  jest.spyOn(fsMock, 'stat').mockImplementation((path, callback) => {
-    callback(null, stats);
-  });
+    /**
+     * Each temp directory will be unique to the test file.
+     * This ensures that temp files/dirs won't cause side effects for other tests.
+     */
+    this.dir = path.resolve(tempDir, 'cosmiconfig', `${relativeParent}-dir`);
 
-  jest.spyOn(fsMock, 'statSync').mockImplementation(() => stats);
-};
+    // create directory
+    makeDir.sync(this.dir);
 
-exports.assertSearchSequence = function assertSearchSequence(
-  readFileMock,
-  searchPaths,
-  startCount
-) {
-  startCount = startCount || 0;
+    (this: any).absolutePath = this.absolutePath.bind(this);
```

Comment:
Would it be better to use and return a closure instead of the class? We could then avoid having to bind the methods.

Reply:
Doesn't really matter to me. I think if a binding is required in the class, it is also required in a closure. That said, I can also just remove the bindings because they aren't required with the current implementations. I've just always considered it best practice to keep them in there to guard against anything unexpected happening.

I guess the "trade-off" with a closure would be another level of indentation or having to return the functions.

Just let me know if I should convert it to a closure, remove the bindings, or leave as is.

Reply:
I'd also be ok with merging this and creating follow-up issues if we want to refactor. I'm hoping that next week I'll have time to get to work on #126, and I'd love to have these test changes merged in before I launch into it.

Reply:
Okay, let's merge this in.